src/main/electron-normalize.ts

```diff
@@ -32,27 +32,117 @@ export function supportsFullProtocol(): boolean {
   return version.major >= 5;
 }
 
+export type ExitReason =
+  | 'clean-exit'
+  | 'abnormal-exit'
+  | 'killed'
+  | 'crashed'
+  | 'oom'
+  | 'launch-failed'
+  | 'integrity-failure';
+
+export const CRASH_REASONS: ExitReason[] = ['crashed', 'oom'];
+export const ALL_REASONS: ExitReason[] = [
+  'clean-exit',
+  'abnormal-exit',
+  'killed',
+  'crashed',
+  'oom',
+  'launch-failed',
+  'integrity-failure',
+];
```

Comment:
```suggestion
export const EXIT_REASONS = [
  'clean-exit',
  'abnormal-exit',
  'killed',
  'crashed',
  'oom',
  'launch-failed',
  'integrity-failure',
] as const;

export type ExitReason = typeof EXIT_REASONS[number];

export const CRASH_REASONS = [EXIT_REASONS[3], EXIT_REASONS[4]] as const;
```

Reply:
Not sure I agree with this one. 

The array indexing makes it easier to break in the future and it's not immediately clear which events are crashes. 

They both transpile down to a similar length JavaScript.

Reply:
yeah I was iffy about the clarity from `CRASH_REASONS` as well. I think the `type ExitReason` is good though.

we can just make it

```ts
export const CRASH_REASONS: Readonly<ExitReason[]> = ['crashed', 'oom'] as const;
```